include/NovelRT/Animation/SpriteAnimatorState.h

```diff
@@ -23,7 +23,7 @@ namespace NovelRT::Animation {
 
       for (size_t i = 0; i < transitionConditions.size(); i++) {
         if (transitionConditions.at(i) == nullptr) {
-          transitionConditions.erase(transitionConditions.begin() + i);
+          transitionConditions.erase(transitionConditions.begin() + static_cast<std::vector<SpriteAnimatorFrame>::iterator::difference_type>(i));
```

Comment:
What's this do?

Reply:
We have a `size_t` which is unsigned, while the difference between two pointers is signed, so we need to convert for this operation to be valid.
* MSVC and Clang warn here because you can overflow going from `unsigned` to `signed` and get potentially undesired behavior

The stdlib has usings in their templated types that tell you what is used for various operations. In this case `iterator::difference_type` which is the equivalent to `ptrdiff_t` for the type contained by the vector.
So we cast our `size_t` to that type.